api-server/server/boot/donate.js

```diff
@@ -243,27 +255,95 @@ export default function donateBoot(app, done) {
       );
   }
+  function addDonation(req, res) {
+    const { user, body } = req;
 
-  const pubKey = keys.stripe.public;
+    if (!user || !body) {
+      return res
+        .status(500)
+        .send({ error: 'User must be signed in for this request.' });
+    }
+    return Promise.resolve(req)
+      .then(
+        user.updateAttributes({
+          isDonating: true
+        })
+      )
+      .then(res.status(200).json({ isDonating: true }))
+      .catch(err => {
+        log(err.message);
+        return res.status(500).send({
+          type: 'danger',
+          message: 'Something went wrong.'
+        });
+      });
+  }
+
+  function updatePaypal(req, res) {
+    const { headers, body } = req;
+    return Promise.resolve(req)
+      .then(getAsyncPaypalToken)
+      .then(token => verifyWebHook(headers, body, token, paypalWebhookId))
+      .then(verifyWebHookType)
+      .then(hookBody => updateUser(hookBody, app))
+      .then(res.status(200).json({ message: 'received hook' }))
+      .catch(err => {
+        log(err.message);
+        return res.status(500).send({
+          type: 'danger',
+          message: 'Something went wrong internally.'
+        });
+      });
+  }
+
+  const stripeKey = keys.stripe.public;
   const secKey = keys.stripe.secret;
+  const paypalKey = keys.paypal.client;
+  const paypalSec = keys.paypal.secret;
   const hmacKey = keys.servicebot.hmacKey;
-  const secretInvalid = !secKey || secKey === 'sk_from_stripe_dashboard';
-  const publicInvalid = !pubKey || pubKey === 'pk_from_stripe_dashboard';
+  const stripeSecretInvalid = !secKey || secKey === 'sk_from_stripe_dashboard';
+  const stripPublicInvalid =
+    !stripeKey || stripeKey === 'pk_from_stripe_dashboard';
+
+  const paypalSecretInvalid =
+    !paypalKey || paypalKey === 'id_from_paypal_dashboard';
+  const paypalPublicInvalid =
+    !paypalSec || paypalSec === 'secret_from_paypal_dashboard';
   const hmacKeyInvalid =
     !hmacKey || hmacKey === 'secret_key_from_servicebot_dashboard';
-
-  if (secretInvalid || publicInvalid || hmacKeyInvalid) {
+  const paypalInvalid = paypalPublicInvalid || paypalSecretInvalid;
+  const stripeInvalid = stripeSecretInvalid || stripPublicInvalid;
+  if (stripeInvalid) {
     if (process.env.FREECODECAMP_NODE_ENV === 'production') {
       throw new Error('Stripe API keys are required to boot the server!');
     }
     console.info('No Stripe API keys were found, moving on...');
-    done();
   } else {
     api.post('/charge-stripe', createStripeDonation);
     api.post('/create-hmac-hash', createHmacHash);
-    donateRouter.use('/donate', api);
-    app.use(donateRouter);
-    app.use('/internal', donateRouter);
+  }
+  if (paypalInvalid) {
+    if (process.env.FREECODECAMP_NODE_ENV === 'production') {
+      throw new Error('PayPal API keys are required to boot the server!');
+    }
+    console.info('No PayPal API keys were found, moving on...');
+  } else {
+    api.post('/update-paypal', updatePaypal);
```

Comment:
This endpoint is not being called yet and so the user record does not get updated.  I think it would make sense if this was called by the client and then, if the user update succeeds, a new donation record should be created (via the logic that's currently being triggered by `add-donation`)

Reply:
so you should tell paypal to send the webhooks to a specific endpoint. I have it setup to my device at the moment but we will be able to point it to .dev and .org for testing.

Reply:
Good point. I completely misunderstood this again.  Sorry about that!

So, for QAing, can we set this up on .dev with this PR and with `FREECODECAMP_NODE_ENV`  set to `development`?

Reply:
Yes we just need to update the donation settings in the config and change the keys in the env